hibernate-core/src/main/java/org/hibernate/dialect/FirebirdDialect.java

```diff
@@ -198,18 +237,97 @@ public void initializeFunctionRegistry(QueryEngine queryEngine) {
 		CommonFunctionFactory.reverse( queryEngine );
 		CommonFunctionFactory.bitandorxornot_binAndOrXorNot( queryEngine );
 		CommonFunctionFactory.leastGreatest_minMaxValue( queryEngine );
-		CommonFunctionFactory.stddevPopSamp( queryEngine );
-		CommonFunctionFactory.varPopSamp( queryEngine );
-		//TODO: lots more statistical functions
-
-		//TODO: gen_uid() and friends, gen_id()
 
-		queryEngine.getSqmFunctionRegistry().registerBinaryTernaryPattern(
+		SqmFunctionRegistry functionRegistry = queryEngine.getSqmFunctionRegistry();
+		functionRegistry.registerBinaryTernaryPattern(
 				"locate",
 				StandardBasicTypes.INTEGER,
 				"position(?1 in ?2)",
-				"(position(?1 in substring(?2 from ?3)) + (?3) - 1)"
-		).setArgumentListSignature("(pattern, string[, start])");
+				"position(?1, ?2, ?3)"
+		).setArgumentListSignature( "(pattern, string[, start])" );
+		functionRegistry.namedDescriptorBuilder( "ascii_val" )
+				.setExactArgumentCount( 1 )
+				.setInvariantType( StandardBasicTypes.SHORT )
+				.register();
+		functionRegistry.registerAlternateKey( "ascii", "ascii_val" );
+		functionRegistry.namedDescriptorBuilder( "ascii_char" )
+				.setExactArgumentCount( 1 )
+				.setInvariantType( StandardBasicTypes.CHARACTER )
+				.register();
+		functionRegistry.registerAlternateKey( "chr", "ascii_char" );
+		functionRegistry.registerAlternateKey( "char", "ascii_char" );
+		functionRegistry.registerPattern(
+				"radians",
+				"((?1)*PI()/180e0)",
```

Comment:
is there a reason to write π in shouty-case here?

Reply:
No reason, will fix it.

Reply:
Thanks :-) Sorry for being anal, it just saves me going through later and lowercasing things.